test/ctia/http/routes/common_test.clj

```diff
@@ -72,7 +76,85 @@
                                          :title "firefox exploit"
                                          :fields ["title"]
                                          :sort_by "disposition"
-                                         :sort_order :desc}))))))
+                                         :sort_order :desc})))
+      (testing "make-date-range-fn should be properly called"
+        (is (= {:date-range {:timestamp
+                             {:gte (tc/from-string "2050-01-01")
+                              :lt "2100-01-01"}}}
+                (sut/search-query :timestamp
+                                  {:from from
+                                   :to to}
+                                  (fn [from to]
+                                    {:gte (tc/from-string "2050-01-01")
+                                     :lt "2100-01-01"}))))))))
+
+(deftest format-agg-result-test
+  (let [from (tc/from-string "2019-01-01")
+        to (tc/from-string "2020-12-31")
+        cardinality 5
+        topn [{:key "Open" :value 8}
+              {:key "New" :value 4}
+              {:key "Closed" :value 2}]
+        histogram [{:key "2020-04-01" :value 3}
+                   {:key "2020-04-02" :value 0}
+                   {:key "2020-04-03" :value 0}
+                   {:key "2020-04-04" :value 6}
+                   {:key "2020-04-05" :value 5}]]
+    (testing "should properly format aggregation results, nested fields and avoid nil filters."
+      (is (= {:data {:observable {:type cardinality}}
+              :type :cardinality
+              :filters {:from from
+                        :to to
+                        :query-string "baddomain"
+                        :field1 "value1"
+                        :field2 "value2"}}
+             (sut/format-agg-result cardinality
+                                    :cardinality
+                                    "observable.type"
+                                    {:date-range
+                                     {:timestamp {:gte from
+                                                  :lt to}}
+                                     :query-string "baddomain"
+                                     :filter-map {:field1 "value1"
+                                                  :field2 "value2"}})))
```

Comment:
I think you should check what occurs if `query-string` contains special chars. 
and the same for `:field*`. I would try `:field1 "foo/bar"`, `"123abc"`, `"abc def"`, etc...